src/client/java/daos-java/src/main/java/io/daos/obj/IODataDesc.java

```diff
@@ -209,20 +209,29 @@ public int getTotalRequestSize() {
     return totalRequestSize;
   }
 
-//  /**
-//   * duplicate this object and all its entries.
-//   * Do not forget to release this object and its entries.
-//   *
-//   * @return duplicated IODataDesc
-//   * @throws IOException
-//   */
-//  public IODataDesc duplicate() throws IOException {
-//    List<Entry> newEntries = new ArrayList<>(akeyEntries.size());
-//    for (Entry e : akeyEntries) {
-//      newEntries.add(e.duplicate());
-//    }
-//    return new IODataDesc(dkey, newEntries, updateOrFetch);
-//  }
+  /**
+   * duplicate this object and all its entries if it's non-reusable desc.
+   * Reusable desc should not call this method. Otherwise UnsupportedOperationException will be thrown.
+   * Do not forget to release this object and its entries.
+   *
+   * @return duplicated IODataDesc
+   * @throws IOException
+   * @throws UnsupportedOperationException
+   */
+  public IODataDesc duplicate() throws IOException {
+    if (isReusable()) {
+      throw new UnsupportedOperationException("reusable desc cannot be duplicated");
+    }
+    IODataDesc dup = new IODataDesc(dkey, IodType.ARRAY, recordSize, updateOrFetch);
+    for (Entry e : akeyEntries) {
+      if (updateOrFetch) {
```

Comment:
It is a little confusing here. The variable name is updateOrFetch. But if it is ture, you only add entry for update. And you add entry for fetch in the else clause.

Reply:
The object can be used for both update and fetch so that code can be reused. 'updateOrFetch' is to control and indicate what this object is for. true for update, false for fetch.